### PR TITLE
Per block

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

(Write your motivation for proposed changes here.)

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/diem/diem/tree/main/developers.diem.com, and link to your PR here.)

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: execution/executor/src/components/apply_chunk_output.rs [hunk 1]
```diff
@@ -153,124 +178,35 @@ impl ApplyChunkOutput {
...
+            let (write_set, events, reconfig_events, gas_used, status) = txn_output.unpack();
             let event_tree = {
-                let event_hashes: Vec<_> =
-                    txn_output.events().iter().map(CryptoHash::hash).collect();
+                let event_hashes: Vec<_> = events.iter().map(CryptoHash::hash).collect();
                 InMemoryAccumulator::<EventAccumulatorHasher>::from_leaves(&event_hashes)
             };
 
-            let txn_info = match txn_output.status() {
+            let state_change_hash = CryptoHash::hash(&write_set);
```
Comment (msmouse) [inline execution/executor/src/components/apply_chunk_output.rs:201]: pointer

Comment (davidiw) [reply execution/executor/src/components/apply_chunk_output.rs:201]: ah, that's true, each state_change_hash is an accumulation on previous state changes, so there's no need to compute a per block state change hash.

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

**Comment by libra-action** [comment]
**release binaries dependency change summary**:
```
target packages:
  M rocksdb 0.17.0 (direct third-party, crates.io)
    * added features: multi-threaded-cf
    * (unchanged features: lz4)
```

File: consensus/consensus-types/src/block.rs [hunk 2]
```diff
@@ -298,6 +298,7 @@ impl Block {
                     .cloned()
                     .map(Transaction::UserTransaction),
             )
+            .chain(std::iter::once(Transaction::StateCheckpoint))
```
Comment (msmouse) [inline consensus/consensus-types/src/block.rs:301]: pointer

**Pull request closed without merging**